src/core/server/saved_objects/import/saved_objects_importer.ts

```diff
@@ -53,6 +55,15 @@ export class SavedObjectsImporter {
     this.#savedObjectsClient = savedObjectsClient;
     this.#typeRegistry = typeRegistry;
     this.#importSizeLimit = importSizeLimit;
+    this.#importHooks = typeRegistry.getAllTypes().reduce((hooks, type) => {
+      if (type.management?.onImport) {
+        return {
+          ...hooks,
+          [type.name]: [type.management.onImport],
```

Comment:
I had to do the following change locally (with TS type failures) to get your test scenario working for dashboards. Is that normal?

```suggestion
      if (type.onImport) {
        return {
          ...hooks,
          [type.name]: [type.onImport],
```

Reply:
We decided to move `onImport` under management (https://github.com/elastic/kibana/pull/87996/commits/5e699b1cfddc70137a9f7956f93e1579c7b85f80). I need to update my patch, sorry...